godot-codegen/src/models/domain/enums.rs

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+//! Definition of engine enums/bitfields and related types.
+//!
+//! See also generator/enums.rs for functions related to turning enums into `TokenStream`s.
+
+use crate::util::ident;
+
+use proc_macro2::{Ident, Literal, TokenStream};
+use quote::{quote, ToTokens};
+
+pub struct Enum {
+    pub name: Ident,
+    pub godot_name: String,
+    pub is_bitfield: bool,
+    pub enumerators: Vec<Enumerator>,
+}
+
+impl Enum {
+    /// Which derives should be implemented for this enum.
+    pub fn derives(&self) -> Vec<Ident> {
+        let mut derives = vec!["Copy", "Clone", "Eq", "PartialEq", "Hash", "Debug"];
+
+        if self.is_bitfield {
+            derives.push("Default");
+        }
+
+        derives.into_iter().map(ident).collect()
+    }
+
+    /// The type we use to represent values of this enum.
+    pub fn ord_type(&self) -> Ident {
+        if self.is_bitfield {
+            ident("u64")
+        } else {
+            ident("i32")
+        }
+    }
+
+    /// Returns all unique enumerator ords, sorted.
+    ///
+    /// Returns `None` if `self` is a bitfield.
+    pub fn unique_ords(&self) -> Option<Vec<i32>> {
+        let mut unique_ords = self
+            .enumerators
+            .iter()
+            .map(|enumerator| enumerator.value.as_enum())
+            .collect::<Option<Vec<i32>>>()?;
```

Comment:
Will this return an `Option` as soon as any element is `None`?

If yes that's fine (there's also [`filter_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map) + explicit check otherwise).

Reply:
Yes, `collect` will fail at the first input, whereas `filter_map` will remove any empty inputs.